translator/conventions/opentelemetry.go

```diff
@@ -0,0 +1,139 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conventions
+
+// OpenTelemetry Semantic Convention values for Resource attribute names.
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+const (
+	AttributeServiceName      = "service.name"
+	AttributeServiceNamespace = "service.namespace"
+	AttributeServiceInstance  = "service.instance.id"
+	AttributeServiceVersion   = "service.version" // proposed
+	AttributeLibraryName      = "library.name"
+	AttributeLibraryLanguage  = "library.language"
+	AttributeLibraryVersion   = "library.version"
+	AttributeContainerName    = "container.name"
+	AttributeContainerImage   = "container.image.name"
+	AttributeContainerTag     = "container.image.tag"
+	AttributeK8sCluster       = "k8s.cluster.name"
+	AttributeK8sNamespace     = "k8s.namespace.name"
+	AttributeK8sPod           = "k8s.pod.name"
+	AttributeK8sDeployment    = "k8s.deployment.name"
+	AttributeHostHostname     = "host.hostname"
+	AttributeHostID           = "host.id"
+	AttributeHostName         = "host.name"
+	AttributeHostType         = "host.type"
+	AttributeCloudProvider    = "cloud.provider"
+	AttributeCloudAccount     = "cloud.account.id"
+	AttributeCloudRegion      = "cloud.region"
+	AttributeCloudZone        = "cloud.zone"
+)
+
+// OpenTelemetry Semantic Convention values for general Span attribute names.
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
+// Peer attributes currently defined in database conventions
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-database.md
+const (
+	AttributeComponent    = "component"
+	AttributePeerAddress  = "peer.address"  // to be obsoleted
+	AttributePeerHost     = "peer.hostname" // to be obsoleted
+	AttributePeerIpv4     = "peer.ipv4"     // to be obsoleted
+	AttributePeerIpv6     = "peer.ipv6"     // to be obsoleted
+	AttributePeerPort     = "peer.port"     // to be obsoleted
+	AttributePeerService  = "peer.service"  // to be obsoleted
+	AttributeNetTransport = "net.transport" // pull request #349
+	TransportTypeTCP      = "IP.TCP"
+	TransportTypeUDP      = "IP.UDP"
+	TransportTypeIP       = "IP"
+	TransportTypeUnix     = "Unix"
+	TransportTypePipe     = "pipe"
+	TransportTypeInProc   = "inproc"
+	TransportTypeOther    = "other"
+	AttributeNetPeerIP    = "net.peer.ip"   // pull request #349
+	AttributeNetPeerPort  = "net.peer.port" // pull request #349
+	AttributeNetPeerName  = "net.peer.name" // pull request #349
+	AttributeNetHostIP    = "net.host.ip"   // pull request #349
+	AttributeNetHostPort  = "net.host.port" // pull request #349
+	AttributeNetHostName  = "net.host.name" // pull request #349
+	AttributeIAMUser      = "iam.user"      // proposed in issue #362
```

Comment:
Let's not define here conventions that are not approved. Feel free to use them in your component's implementation directly if you need them.

Reply:
These need to be added before Otel API v1 is final. I think we should leave and then edit or remove if the API is finalized without them.

Reply:
I disagree. Having a proposal to add them to specification does not make them automatically qualify to be part of this repo. I do not think we should add code that corresponds to unapproved, under review, in-flux proposals to this repo. Once it is added it becomes my responsibility as a maintainer to follow the specification and maintain the code in this repo against a moving target, which is not a task that I am willing to undertake.

To clarify, I am not against the particular proposal and I am not suggesting anything that blocks you. You can equally easily define these constants in your component's code in the contrib repo. This way you take the responsibility to maintain that part of the code. Should the proposal to add this to OpenTelemetry semantic conventions be approved the constants can be moved to this repo in the future.